### DISCUSSION TITLE
I think my TinyPICO V2 (micro USB) is broken

### DISCUSSION THREAD

**Discussion by ejgeiger** [micropython/micropython#13136] [ESP32] [answered]
Hi, my device went down a few days ago.  It was running CircuitPython 8 and I wasn't able to get it to enter REPL on Thonny.  I am able to erase and reflash it, so I downloaded the most current Micropython and got more detailed errors that repeat with the number of failed writes increasing: 

```
ELF file SHA256: 200923fbf6940d61

Rebooting...
ets Jun  8 2016 00:22:57

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 188777542, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:4968
load:0x40078000,len:15860
load:0x40080400,len:3700
entry 0x400805e0
E (983) esp_psram: SPI SRAM memory test fail. 2191/131072 writes failed, first @ 3F800280

E (983) cpu_start: External RAM failed memory test!

abort() was called at PC 0x4008156d on core 0

Backtrace: 0x40082141:0x3ffe3b90 0x4009a049:0x3ffe3bb0 0x4009e069:0x3ffe3bd0 0x4008156d:0x3ffe3c40 0x400796c0:0x3ffe3c80 |<-CORRUPTED

ELF file SHA256: 200923fbf6940d61
```

This leads me to think the RAM chip has died, but before sending this to the silicon graveyard, I wanted to know if there were any other troubleshooting steps?

details:
Thonny 4.14
TinyPICO V2 with microUSB
UM_TINYPICO-20231005-v1.21.0.bin

**Comment by robert-hh** [comment, marked as answer]
Is anything attached to the board, or do you see any broken connections, e.g. along the sides of the ESP32?
If you have a voltmeter you could check the value of the 3V3 power. The ESP32 can run at a slightly less voltage than the RAM chip.
The RAM chip itself comes in a comparable large package, which allows to replace it, if broken.
And even with a broken RAM chip you could load GENERIC_ESP32 firmware to it, ignoring the SPIRAM.

**Reply by ejgeiger** [reply]
GENERIC_ESP32 works.  Thanks for that.  3V3 is rock solid and the board is currently not attached to anything.  I don't think I need that much RAM, so I am not in a hurry to repair it, but will report back when I do.

**Comment by liudr** [comment]
This bad ext. RAM seems to be a reoccurring problem although not very common with later revisions of ESP modules. For instance, earlier with ESP32 rev b (2020-21) I had more issues with external RAM not passing tests than now with rev e. I still got one that has bad ext RAM and two with non-working WiFi, among say, 500 ESP32 modules I've used in the past couple of years. I don't know how Espressif does QC but it doesn't look very good on them when these essential features fail to work.
